lib/hammer_cli_foreman/command_extensions/update_common.rb

```diff
@@ -7,7 +7,12 @@ class UpdateCommon < HammerCLI::CommandExtensions
 
       request_params do |params, command_object|
         update_params = params[command_object.resource.singular_name]
-        command_object.context[:action_message] = :nothing_to_do if update_params && update_params.empty?
+        update_params =  update_params.nil? ? params : update_params.values.select { | val| !val.nil?}
+        command_object.context[:action_message] = :nothing_to_do unless valid?(update_params)
+      end
+
+      def self.valid?(params)
+        params.nil? ? false : !(params.empty? || params.select { | param | param.is_a?(Hash) && param.empty? }.count == params.count)
```

Comment:
Let's assume that we have `params = {"id"=>1}` then `valid?(params)` will be `true` and thus the message won't be applied. So, the commands which allow passing `id` only in `update` commands will still print `success` instead.

Moreover, such commands as `hammer activation-key update --organization-id 1 --id 1` use additional options/params (in this case it's `organization_id`) for search puproses. So the mentioned command won't actually update anything, it will just use both params to find the activation key, but won't change anything.

I guess we need a different approach here. Maybe we could mark somehow options/params which used for change applying and then check if they were set?.. Just an idea, it could also fail.